src/mode/modeHandler.ts

```diff
@@ -1446,7 +1448,7 @@ export class ModeHandler implements vscode.Disposable {
             await this.handleMultipleKeyEvents(keyStrokes);
           }
 
-          vimState.isReplayingMacro = false;
+          vimState.isReplayingMacro --;
```

Comment:
The idea is that macros can be recursive, so we might call this code again inside `             vimState = await this.runMacro(vimState, recordedMacro);
`
vimState is a sort of global state.